src/openforms/variables/validators.py

```diff
@@ -114,5 +120,39 @@ def __call__(self, value: Mapping[str, str] | JSONValue) -> None:
             raise ValidationError(errors)
 
 
+class ServiceFetchConfigurationValidator:
```

Comment:
I'd break this out into more validators that can be composed:

1. `ServiceFetchMethodValidator`
2. `ServiceFetchMappingExpressionValidator`

so that you can properly use the `code` for the validation error and let the errors be combined by DRF.

https://www.django-rest-framework.org/api-guide/validators/#writing-custom-validators

This should also make them more unit-testable.